hello_test.cc

```diff
@@ -1,10 +1,155 @@
-#include <gtest/gtest.h>
-
-// Demonstrate some basic assertions.
-TEST(HelloTest, BasicAssertions) {
-  // Expect two strings not to be equal.
-  EXPECT_STRNE("hello", "world");
-  // Expect equality.
-  EXPECT_EQ(7 * 6, 42);
-}
-
+#include <gtest/gtest.h>
+#include <vector>
+#include <tuple>
+
+#include "element.h"
+#include "board.h"
+#include "movemap.h"
+
+TEST(Sphere, Constructors) {
+  Sphere sphere;
+
+  for (auto element : ELEMENTLIST) {
+    EXPECT_EQ(sphere.at(element), 0);
+  }
+
+  Sphere sphere2({Element::FIRE});
+  for (auto element : ELEMENTLIST) {
+    EXPECT_EQ(sphere2.at(element), element == Element::FIRE);
+  }
+
+  Sphere sphere3({{Element::WATER, 1}, {Element::EARTH, 2}});
+  for (auto element : ELEMENTLIST) {
+    if (element == Element::WATER) {
+      EXPECT_EQ(sphere3.at(element), 1);
+    }
+    else if (element == Element::EARTH)
+    {
+      EXPECT_EQ(sphere3.at(element), 2);
+    }
+    else {
+      EXPECT_EQ(sphere3.at(element), 0);
+    }
+  }
+
+}
+
+TEST(Sphere, CountTest) {
+  Sphere sphere;
+
+  for (auto element : ELEMENTLIST) {
+    EXPECT_EQ(sphere.count(element), 0);
+  }
+
+  Sphere sphere2({Element::FIRE});
+  for (auto element : ELEMENTLIST) {
+    EXPECT_EQ(sphere2.count(element), element == Element::FIRE);
+  }
+
+  Sphere sphere3({{Element::WATER, 1}, {Element::EARTH, 2}});
+  for (auto element : ELEMENTLIST) {
+    if (element == Element::WATER) {
+      EXPECT_EQ(sphere3.count(element), 1);
+    }
+    else if (element == Element::EARTH)
+    {
+      EXPECT_EQ(sphere3.count(element), 2);
+    }
+    else {
+      EXPECT_EQ(sphere3.count(element), 0);
+    }
+  }
+  EXPECT_EQ(sphere3.count(), 3);
+}
+
+
+
+TEST(Board, BoardInitGemCounts) {
+  Board board;
+  
+  for (auto element : ELEMENTLIST) {
+    EXPECT_EQ(board.count(element, element), 1);
+    EXPECT_EQ(board.count(element, ~element), 0);
+  }
+
+}
+
+TEST(Board, MoveMapTest) {
+  std::vector<std::tuple<Element, Element, Element>> moves = {{Element::FIRE, Element::FIRE, Element::WATER}, {Element::EARTH, Element::EARTH, Element::LIGHTNING}, {Element::EARTH, Element::EARTH, Element::LIGHTNING}};
+  
+  MoveMap moveMap(moves);
+  EXPECT_EQ(moveMap.inputCount(), 3);
+  EXPECT_EQ(moveMap.inputCount(Element::EARTH, Element::EARTH), 2);
+  EXPECT_EQ(moveMap.outputCount(), 3);
+  EXPECT_EQ(moveMap.outputCount(Element::LIGHTNING, Element::EARTH), 2);

```

Comment:
Add a newline between 80 and 81. Checks and logic shouldn't be mixed. Keeping them separate makes them easier to read.